src/NovelRT/Windowing/WindowingService.cpp

```diff
@@ -36,13 +36,28 @@ namespace NovelRT::Windowing {
       checkForOptimus(OptimusLibraryName);
 #endif
 
+      
+#if defined(__APPLE__)
+      _logger.logInfoLine("Attempting to create OpenGL 4.1 context");
+      glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
+      glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
+      glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GLFW_TRUE);
+      glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
+      
+      auto window = glfwCreateWindow(wData, hData, windowTitle.c_str(), nullptr, nullptr);
+      if (window == nullptr)
+      {
+          _logger.throwIfNullPtr(window, "Failed to create OpenGL v4.1 context");
+      }
+#else
```

Comment:
If-def'ed a much smaller context creation method for macOS - this is the only context that works with our current calls and shaders and needs to be called this way, so if the target macOS device does not support OGL 4.1 for some reason it will fail straight away. Other OSes will continue to follow the same pattern we normally use.

Reply:
How come we cannot check for other contexts, exactly?

Reply:
Although we may be able to check for a OGL 3.2 context, this would set us back to GLSL version 1.40 approximately and would affect our feature set. Also, EGL Contexts as well as Compatibility contexts are not supported by Mac (from my research, I could be wrong). (And calling the wrong context _immediately_ nopes out the `glfwCreateWindow` call)

Reply:
Ah ok, makes sense.